tests/device_tests/test_msg_applysettings.py

```diff
@@ -159,34 +159,55 @@ def test_apply_homescreen(self, client):
     @pytest.mark.skip_t1
     @pytest.mark.setup_client(pin=None)
     def test_safety_checks(self, client):
-        BAD_ADDRESS = parse_path("m/0")
+        def get_bad_address():
+            btc.get_address(client, "Bitcoin", parse_path("m/0"))
+
+        assert client.features.safety_checks == messages.SafetyCheckLevel.Strict
 
         with pytest.raises(
             exceptions.TrezorFailure, match="Forbidden key path"
         ), client:
-            client.set_expected_responses([messages.Failure()])
-            btc.get_address(client, "Bitcoin", BAD_ADDRESS)
+            get_bad_address()
 
         with client:
             client.set_expected_responses(EXPECTED_RESPONSES_NOPIN)
             device.apply_settings(
-                client, safety_checks=messages.SafetyCheckLevel.Prompt
+                client, safety_checks=messages.SafetyCheckLevel.PromptAlways
             )
 
+        assert client.features.safety_checks == messages.SafetyCheckLevel.PromptAlways
+
         with client:
             client.set_expected_responses(
                 [messages.ButtonRequest(), messages.Address()]
             )
-            btc.get_address(client, "Bitcoin", BAD_ADDRESS)
+            get_bad_address()
 
         with client:
             client.set_expected_responses(EXPECTED_RESPONSES_NOPIN)
             device.apply_settings(
                 client, safety_checks=messages.SafetyCheckLevel.Strict
             )
 
+        assert client.features.safety_checks == messages.SafetyCheckLevel.Strict
+
         with pytest.raises(
             exceptions.TrezorFailure, match="Forbidden key path"
         ), client:
-            client.set_expected_responses([messages.Failure()])
```

Comment:
please keep the expected responses

(they ensure that the `Failure` is the first thing to happen, as opposed to, e.g., after some dialog)

Reply:
Fixed, thanks.